### DISCUSSION TITLE
Spectacular Application Harness for integration of entire Angular app

### DISCUSSION THREAD

**Discussion by meirka** [ngworker/ngworker#35] [Q&A]
Do I use ```createApplicationHarness({providers: [applicationInitializer]});``` in order to test entire app, including forms, navigation backend calls?

**Comment by LayZeeDK** [comment]
Try adding your AppModule to the imports option. However, Spectacular's Application testing API is currently only officially supporting integration testing of bootstrap listeners and application initializers. Any capabilities beyond this are happy coincidences.

If your test case primarily exercises one part of your application (page/routed feature/domain), please try Spectacular's Feature testing API instead. It support everything you're requesting but scopes the tests setup to a single routed feature module and everything encapsulated by it out-of-the-box.
